cni/network/invoker_cns.go

```diff
@@ -356,3 +304,158 @@ func (invoker *CNSIPAMInvoker) Delete(address *net.IPNet, nwCfg *cni.NetworkConf
+		}
+
+		if net.ParseIP(info.podIPAddress).To4() != nil { //nolint:gocritic
+			ncgw, err = getOverlayGateway(ncIPNet)
+			if err != nil {
+				return err
+			}
+		} else if net.ParseIP(info.podIPAddress).To16() != nil {
+			ncgw = net.ParseIP(overlayGatewayV6IP)
+		} else {
+			return errors.Wrap(err, "No podIPAddress is found: %w")
+		}
+	}
+
+	// construct ipnet for result
+	resultIPnet := net.IPNet{
+		IP:   ip,
+		Mask: ncIPNet.Mask,
+	}
+
+	if ip := net.ParseIP(info.podIPAddress); ip != nil {
+		defaultInterfaceInfo := addResult.defaultInterfaceInfo.ipResult
+		if defaultInterfaceInfo == nil {
+			defaultInterfaceInfo = &cniTypesCurr.Result{}
+		}
+
+		defaultRouteDstPrefix := network.Ipv4DefaultRouteDstPrefix
+		if ip.To4() == nil {
+			defaultRouteDstPrefix = network.Ipv6DefaultRouteDstPrefix
+			addResult.ipv6Enabled = true
+		}
+
+		defaultInterfaceInfo.IPs = append(defaultInterfaceInfo.IPs,
+			&cniTypesCurr.IPConfig{
+				Address: resultIPnet,
+				Gateway: ncgw,
+			})
+
+		routes, err := getRoutes(info.routes)
+		if err != nil {
+			return err
+		}
+
+		if len(routes) > 0 {
+			defaultInterfaceInfo.Routes = append(defaultInterfaceInfo.Routes, routes...)
+		} else { // add default routes if none are provided
+			defaultInterfaceInfo.Routes = append(defaultInterfaceInfo.Routes, &cniTypes.Route{
+				Dst: defaultRouteDstPrefix,
+				GW:  ncgw,
+			})
+		}
+
+		addResult.defaultInterfaceInfo.ipResult = defaultInterfaceInfo
+		addResult.defaultInterfaceInfo.skipDefaultRoutes = info.skipDefaultRoutes
+	}
+
+	// get the name of the primary IP address
+	_, hostIPNet, err := net.ParseCIDR(info.hostSubnet)
+	if err != nil {
+		return fmt.Errorf("unable to parse hostSubnet: %w", err)
+	}
+
+	addResult.hostSubnetPrefix = *hostIPNet
+	addResult.defaultInterfaceInfo.nicType = cns.InfraNIC
+
+	// set subnet prefix for host vm
+	// setHostOptions will execute if IPAM mode is not v4 overlay and not dualStackOverlay mode
+	// TODO: Remove v4overlay and dualstackoverlay options, after 'overlay' rolls out in AKS-RP
+	if !overlayMode {
+		if err := setHostOptions(ncIPNet, addConfig.options, info); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func configureSecondaryAddResult(info *IPResultInfo, addResult *IPAMAddResult, podIPConfig *cns.IPSubnet) error {
+	ip, ipnet, err := podIPConfig.GetIPNet()
+	if ip == nil {
+		return errors.Wrap(err, "Unable to parse IP from response: "+info.podIPAddress+" with err %w")
+	}
+
+	macAddress, err := net.ParseMAC(info.macAddress)
+	if err != nil {
+		return errors.Wrap(err, "Invalid mac address")
+	}
+
+	result := InterfaceInfo{
+		ipResult: &cniTypesCurr.Result{
+			IPs: []*cniTypesCurr.IPConfig{
+				{
+					Address: net.IPNet{
+						IP:   ip,
+						Mask: ipnet.Mask,
+					},
+				},
+			},
+			Interfaces: []*cniTypesCurr.Interface{
+				{
+					Mac: info.macAddress,
+				},
+			},
+		},
+		nicType:           cns.DelegatedVMNIC,
+		macAddress:        macAddress,
```

Comment:
Why do we need this if `cniTypesCurr.Interface` has `Mac` field?

Reply:
removed

Reply:
is this removed? Why am i still seeing this code then? my ask is why can't we reuse `cniTypesCurr.Interface` and avoid adding `macAddress` field separately

Reply:
we removed the cniTypesCurr.Interface and kept macAddress field since the one in cniTypesCurr.Interface is a string while we need the parsed macAddress for endpointInfo

Reply:
but why cant we reuse `cniTypesCurr.Result.Interface` structure? then no need for new field `macAddress`

Reply:
macAddress in cniTypesCurr.Result.Interface is string but we need net.HardwareAddr for EndpointInfo, just passing the net.HardwareAddr as new field since we already parsed mac here